packages/vscode-extension/src/builders/buildAndroid.ts

```diff
@@ -69,10 +75,10 @@ export async function buildAndroid(
   cancelToken: CancelToken,
   outputChannel: OutputChannel,
   progressListener: (newProgress: number) => void
-) {
+): Promise<AndroidBuildResult> {
   if (await isExpoGoProject()) {
     const apkPath = await downloadExpoGo(DevicePlatform.Android, cancelToken);
-    return { apkPath, packageName: EXPO_GO_PACKAGE_NAME };
+    return { apkPath, packageName: EXPO_GO_PACKAGE_NAME, platform: DevicePlatform.Android };
```

Comment:
We need to establish some tag to check the difference between iOS and Android package info.
This code was clunky from the start and even after refactor isn't still right.

The main problem here comes from the fact that keys are different – `apkPath` vs `appPath`, yet we want to consume them the same way.
We could get rid of `platform` and check for specific keys but it's less intuitive that way. The real solution is to unify keys instead of having union of two objects with same content but different keys.

I didn't want to make this refactor bigger so I opted for this solution.

Reply:
Would you mind adding a comment then with TODO explaining the end version you would like to see implemented?